File: src/main/java/burrows/apps/gif/example/di/module/LeakCanaryModule.java
package burrows.apps.gif.example.di.module;

import android.app.Application;
import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public class LeakCanaryModule {
  private final Application application;

  public LeakCanaryModule(Application application) {
    this.application = application;
  }

  @Provides @Singleton RefWatcher provideRefWatcher() {
    return LeakCanary.install(application);
  }
}

File: src/main/kotlin/burrows/apps/example/gif/presentation/di/module/LeakCanaryModule.kt
package burrows.apps.example.gif.presentation.di.module

import android.app.Application
import burrows.apps.example.gif.presentation.di.scope.PerActivity
import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher
import dagger.Module
import dagger.Provides

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
@Module
class LeakCanaryModule {
  @Provides @PerActivity fun providesRefWatcher(application: Application): RefWatcher = LeakCanary.install(application)
}
